front/src/components/boxs/device-in-room/DevicesBox.jsx

```diff
@@ -1,34 +1,47 @@
 import { Component } from 'preact';
 import { connect } from 'unistore/preact';
+import Promise from 'bluebird';
 import get from 'get-value';
 import { RequestStatus } from '../../../utils/consts';
-import { WEBSOCKET_MESSAGE_TYPES } from '../../../../../server/utils/constants';
+import {
+  WEBSOCKET_MESSAGE_TYPES,
+  DEVICE_FEATURE_CATEGORIES,
+  DEVICE_FEATURE_TYPES
+} from '../../../../../server/utils/constants';
 import DeviceCard from './DeviceCard';
 import debounce from 'debounce';
 
-const updateDevices = (devices, deviceFeatureSelector, lastValue, lastValueChange) => {
-  return devices.map(device => {
-    return {
-      ...device,
-      features: device.features.map(feature => {
-        if (feature.selector === deviceFeatureSelector) {
-          return {
-            ...feature,
-            last_value: lastValue,
-            last_value_changed: lastValueChange
-          };
-        }
-        return feature;
-      })
-    };
+const updateDevices = (deviceFeatures, deviceFeatureSelector, lastValue, lastValueChange) => {
+  return deviceFeatures.map(feature => {
+    if (feature.selector === deviceFeatureSelector) {
+      return {
+        ...feature,
+        last_value: lastValue,
+        last_value_changed: lastValueChange
+      };
+    }
+    return feature;
+  });
+};
+
+const updateDevicesString = (deviceFeatures, deviceFeatureSelector, lastValueString, lastValueChange) => {
```

Comment:
You need to change the name of this function because you don't update devices but deviceFeatures